### PR TITLE
Create modal for foster me

### LINKED ISSUE #103: [Foster offer] [FE] Offer form dynamic question framework
## Context

We want users to answer a series of questions when they click the "Foster Me" button on the post view. The questions that are posed to the volunteer are different based on which `type` the post is. We want to be able to easily modify the questions according to the needs of Angels Among Us, which necessitates building a "framework" of sorts to define the question data. This ticket is the first step of 2 tickets that builds out the entire frontend for the foster me flow.

## Requirements

First, you are to create the modal that pops up when a user clicks on "Foster Me". This can be blank for now. For details on how these modal popups should look, reach out to @katsukixyz and @winniezhangg. Define a data structure by which the data in the questionnaire is stored. My recommendation is something like:

```js
{
  [FosterType.Return]: [
    {
      key: "numOtherDogs",
      title: "How many other fosters and personal dogs do you have?"
    }
    ...
  ],
  [FosterType.Boarding]: [ ... ],
  ...
}
```

The input boxes to all these questions can all be made textarea inputs (will be built in a later ticket).

Based on the data structure you define, write some logic to retrieve the correct questions according to a foster type. You can hard-code the foster type for now since #94 will handle passing the props into the post view. The modal you create should be passed props that hold the information about a post.

As far as things that are **out of scope** for this ticket, you **do not** need to implement the design for the modals. This will be handled by the next ticket in building the foster me flow. In addition, you will need to build validation around the textarea inputs, but this will be handled later. You can focus on building out this framework for this ticket.

#### PM

- [ ] A modal pops up when I click the "Foster Me" button
- [ ] These modals have space/account for there to be several questions and text responses
- [ ] The modals match the design and are dynamically sized (spacing/layout isn't hard coded)
- [ ] Depending on the type of dog, different modals can/will pop up
- [ ] Clicking outside of the modal closes the modal
- [ ] Mobile view experience is equivalent to desktop

#### EM

- [ ] Data structure defined is intuitive, and allows for easily editing questions as client needs change
- [ ] Modals are passed the relevant props needed to properly display the information required by the design

#### Designer

- [ ] Top part with title is fixed, but the framed area with questions should be scrolled with submit at the end
- [ ] Check out Figma prototype under pet post desktop 2 to see the functionality

### PR DESCRIPTION
**Pull request by its-edmund**
## PR Title

Issue Number(s): #number_of_issue.

What does this PR change and why?

### Checklist

- [ ] Requirements have been implemented
- [ ] Acceptance criteria are met
- [ ] Database schema docs have been updated or are not necessary
- [ ] Code follows design and style guidelines
- [ ] Commits follow guidelines (concise, squashed, etc)
- [x] Relevant reviewers (Senior Dev/EM/Designers) have been assigned to this PR

### Critical Changes

- Database change/migration to run
- Environment config change
- Breaking API change

### Related PRs

- #number_of_pr

### Testing

Enumerate steps to test the functionality of your ticket. This should include edge cases and testing of error handling, if applicable.

### CODE REVIEW

File: web/components/PetPostModal/PetPostModal.tsx [hunk 1]
```diff
@@ -3,18 +3,199 @@ import "react-responsive-carousel/lib/styles/carousel.min.css";
...
+          height="100vh"
+          display={["flex", "none"]}
+        >
+          <Stack
+            marginTop={4}
+            marginLeft={4}
+            direction="row"
+            onClick={onFormViewClose}
+            spacing={2}
+            color="#7D7E82"
```
Comment (katsukixyz) [inline web/components/PetPostModal/PetPostModal.tsx]: Nit: use `text-secondary`

File: web/components/PetPostModal/PetPostModal.tsx [hunk 2]
```diff
@@ -3,18 +3,199 @@ import "react-responsive-carousel/lib/styles/carousel.min.css";
...
+                <Flex key={question.key} flexDir="column" gap={2}>
+                  <Text>{question.title}</Text>
+                  <Input />
+                </Flex>
+              );
+            })}
+          </Flex>
+        </Flex>
+        <ModalFooter display={["none", "flex"]}>
+          <Button variant="ghost" mr={3} onClick={onFormViewClose}>
```
Comment (katsukixyz) [inline web/components/PetPostModal/PetPostModal.tsx]: Use `variant="outline-secondary"`

**Review by katsukixyz — APPROVED** [review]
Looks good, left a few nits. Sorry for the late review 😭

**Comment by winniezhangg** [comment]
Great work! Couple of things:
- Can the margins be increased/ increase the overall size of the modal? 
- Fix the 'Cancel' and 'Submit' to the bottom of the form
